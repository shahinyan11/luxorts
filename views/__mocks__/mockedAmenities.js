export default {
  'Amina Heaney': {
    '5e066ad7-4194-4bd4-95ec-0d9715f8159b': {
      id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
      description: 'Maiores porro nostrum dolorum sunt.',
      name: 'Nolan Kertzmann',
      status: 'active',
      category: null,
      checked: false,
    },
  },
  'Torrie Sawayn': {
    'c277033f-1683-4d2d-a409-d00b0bbe8201': {
      id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
      description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
      name: 'Stephine Rohan',
      status: 'active',
      category: null,
      checked: false,
    },
  },
};
