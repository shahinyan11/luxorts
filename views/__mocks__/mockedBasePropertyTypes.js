export default [
  {
    value: '51e35121-c27e-46fc-9feb-b51dab2490ee',
    label: 'repudiandae',
    description: 'description repudiandae',
  },
  {
    value: '995c5714-185d-4085-8512-690903334f6e',
    label: 'molestias',
    description: 'description molestias',
  },
];
