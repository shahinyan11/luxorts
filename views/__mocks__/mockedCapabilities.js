export default {
  APPROVED: {
    '11da7909-9f4c-45d6-a2a0-8a105b067ed2': {
      id: '11da7909-9f4c-45d6-a2a0-8a105b067ed2',
      description: 'Vel commodi ad incidunt repellendus a molestias delectus nam.',
      name: 'Tonya Wunsch',
      status: 'active',
      checked: false,
    },
    'b84c898e-2176-4ac8-9dfa-54186ccef6cb': {
      id: 'b84c898e-2176-4ac8-9dfa-54186ccef6cb',
      description: 'Maxime nostrum ex iste velit nulla incidunt.',
      name: 'Edwin Greenholt',
      status: 'active',
      checked: false,
    },
  },
};
