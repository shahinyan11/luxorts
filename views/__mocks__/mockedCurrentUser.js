export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
