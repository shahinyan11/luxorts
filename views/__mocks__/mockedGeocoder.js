export default [
  {
    streetNumber: '240',
    street: '240 North 7th Avenue',
    city: 'Phoenix',
    state: 'Arizona',
    country: 'United States',
    zipCode: '85007',
    fullAddress: '240 N 7th Ave, Phoenix, AZ 85007, USA',
    longitude: -112.0825682,
    latitude: 33.4507026,
  },
];
