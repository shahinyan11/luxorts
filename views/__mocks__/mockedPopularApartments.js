export default [
  {
    id: '06a30f07-23b5-4900-b83c-b56986f119bc',
    bedroomsAmount: 1,
    bathsAmount: 28,
    propertyTypeMame: 'voluptatum',
    title: 'Alta Kling',
    guestsNumber: 3,
    bedsAmount: 10,
    description: 'Vanessa Swift',
    typeOfPlace: 'shared_room',
    pricePerDay: 6860000,
    limitedListingPhotos: [
      {
        id: '115e2205-856f-4ed4-9f17-1dd6f37f7520',
        photoUrls: {
          small:
            'https://luxorts-backend-staging.s3.us-east-2.amazonaws.com/uploads/listingphoto/115e2205-856f-4ed4-9f17-1dd6f37f7520/photo/small-cf3518983b1ecf97422a2bda10eca45b.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAXWPQSY4555KWNWFP%2F20220717%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20220717T094425Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=128ecc291ddb112fec4d57ff3e1fb7b91249045e54e653d8b083113e9b9115a7',
        },
      },
    ],
  },
];
