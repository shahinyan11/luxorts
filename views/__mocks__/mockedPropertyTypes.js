export default [
  {
    id: 'd46e3bec-bb9a-4574-ba6d-1a3e19332b5a',
    name: 'totam',
    listingsCount: 1,
  },
];
