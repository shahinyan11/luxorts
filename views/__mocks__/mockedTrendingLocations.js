export default [
  {
    id: 'united-states-pennsylvania-mineral-point',
    country: 'United States',
    city: 'Baltimore',
    state: 'Maryland',
  },
];
