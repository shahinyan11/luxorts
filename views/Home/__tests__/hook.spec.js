import { renderHook } from '@testing-library/react-hooks';

import { dispatch, store } from '__mocks__/react-redux';

import {
  fetchHomePageData,
  fetchPopularApartments,
  setApartmentsPage,
} from 'state/concepts/publicInfo/actions';

import mockedPopularApartments from 'views/__mocks__/mockedPopularApartments';
import mockedPropertyTypes from 'views/__mocks__/mockedPropertyTypes';
import mockedTrendingLocations from 'views/__mocks__/mockedTrendingLocations';

import useContainer, { getInitialProps } from '../hook';

const selfPage = 1;

jest.mock('state/concepts/publicInfo/selectors', () => ({
  propertyTypesSelector: jest.fn(() => mockedPropertyTypes),
  trendingLocationsSelector: jest.fn(() => mockedTrendingLocations),
  popularApartmentsLastPageSelector: jest.fn(() => 7),
  popularApartmentsSelfPageSelector: jest.fn(() => selfPage),
}));

jest.mock('state/concepts/listings/selectors', () => ({
  listingsSelector: jest.fn(() => mockedPopularApartments),
}));

describe('Home useContainer hook', () => {
  let result = null;

  beforeEach(() => {
    ({ result } = renderHook(useContainer));
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('handlePrev method must dispatch setApartmentsPage and fetchPopularApartments actions', () => {
    result.current.handlePrev();

    expect(dispatch).toHaveBeenCalledWith(setApartmentsPage(selfPage - 1));
    expect(dispatch).toHaveBeenCalledWith(fetchPopularApartments());
  });

  it('handleNext method must dispatch setApartmentsPage and fetchPopularApartments actions', () => {
    result.current.handleNext();

    expect(dispatch).toHaveBeenCalledWith(setApartmentsPage(selfPage + 1));
    expect(dispatch).toHaveBeenCalledWith(fetchPopularApartments());
  });

  describe('getInitialProps method', () => {
    const ctx = { store };

    it('dispatches actions', async () => {
      await getInitialProps(ctx);

      expect(dispatch).toHaveBeenCalledWith(fetchHomePageData());
    });
  });
});
