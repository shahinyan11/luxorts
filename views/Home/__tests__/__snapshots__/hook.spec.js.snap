// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home useContainer hook matches snapshot 1`] = `
Object {
  "formik": Object {
    "dirty": true,
    "handleSubmit": [MockFunction],
    "initialValues": Object {
      "date": "",
      "location": "",
    },
    "isValid": true,
    "resetForm": [MockFunction],
    "setFieldTouched": [MockFunction],
    "setFieldValue": [MockFunction],
    "setValues": [MockFunction],
    "validateForm": [MockFunction],
    "values": Object {},
  },
  "handleNext": [Function],
  "handlePrev": [Function],
  "popularApartments": Array [
    Object {
      "bathsAmount": 28,
      "bedroomsAmount": 1,
      "bedsAmount": 10,
      "description": "Vanessa Swift",
      "guestsNumber": 3,
      "id": "06a30f07-23b5-4900-b83c-b56986f119bc",
      "limitedListingPhotos": Array [
        Object {
          "id": "115e2205-856f-4ed4-9f17-1dd6f37f7520",
          "photoUrls": Object {
            "small": "https://luxorts-backend-staging.s3.us-east-2.amazonaws.com/uploads/listingphoto/115e2205-856f-4ed4-9f17-1dd6f37f7520/photo/small-cf3518983b1ecf97422a2bda10eca45b.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAXWPQSY4555KWNWFP%2F20220717%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20220717T094425Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=128ecc291ddb112fec4d57ff3e1fb7b91249045e54e653d8b083113e9b9115a7",
          },
        },
      ],
      "pricePerDay": 6860000,
      "propertyTypeMame": "voluptatum",
      "title": "Alta Kling",
      "typeOfPlace": "shared_room",
    },
  ],
  "popularApartmentsLastPage": 7,
  "popularApartmentsSelfPage": 1,
  "propertyTypes": Array [
    Object {
      "id": "d46e3bec-bb9a-4574-ba6d-1a3e19332b5a",
      "listingsCount": 1,
      "name": "totam",
    },
  ],
  "trendingLocations": Array [
    Object {
      "city": "Baltimore",
      "country": "United States",
      "id": "united-states-pennsylvania-mineral-point",
      "state": "Maryland",
    },
  ],
}
`;
